src/js/settings/components/provider-settings/ollama-base.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalInputControl as InputControl, // eslint-disable-line @wordpress/no-unsafe-wp-apis
+	SelectControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { SettingsRow } from '../settings-row';
+
+/**
+ * Component for the base Ollama Provider settings.
+ *
+ * This component is used within the ProviderSettings component to
+ * allow users to configure the Ollama Provider settings.
+ *
+ * @param {Object}   props                  Component props.
+ * @param {Object}   props.providerSettings The provider settings.
+ * @param {Function} props.onChange         Function to call when the provider settings change.
+ * @param {string}   props.providerName     Name of Provider.
+ * @return {React.ReactElement} OllamaBaseSettings component.
+ */
+export const OllamaBaseSettings = ( {
+	providerSettings,
+	onChange,
+	providerName,
+} ) => {
+	const Description = () => (
+		<>
+			{ __(
+				'URL of the locally hosted Ollama instance. Defaults to http://localhost:11434/. ',
+				'classifai'
+			) }
+			{ __( "Don't have Ollama installed yet? ", 'classifai' ) }
+			<a
+				title={ __( 'Install Ollama', 'classifai' ) }
+				href="https://ollama.com/"
+				target="_blank"
+				rel="noopener noreferrer"
+			>
+				{ __( 'Download Ollama', 'classifai' ) }
+			</a>
+		</>
+	);
+
+	const models = [
+		{ label: __( '-- Choose Model --', 'classifai' ), value: '' },
+	];
+
+	// Convert providerSettings.models to an array from an object.
+	if (
+		providerSettings?.models &&
+		! Array.isArray( providerSettings.models )
+	) {
+		for ( const [ key, value ] of Object.entries(
+			providerSettings.models
+		) ) {
+			models.push( { label: value, value: key } );
+		}
+	}
```

Comment:
Optional note: How about loading models by making an API call if provider settings don't have models saved or the endpoint URL is updated by the user in settings? This way, we don't need to save settings to populate models during first-time setup.

Reply:
Ah, sorry, missed this in fixing a few other things. I do think either a refresh button or maybe just always pulling in the current models when on the settings page would be a nice enhancement here. Would make it easier to see new models after you install them